3-dynamic-analysis/3-profiling/1-benchmarks.md

```diff
@@ -161,21 +161,121 @@ Sometimes your benchmarks might be way faster that you would expect. When
 that happens do not be too happy, it might be caused by a compiler optimization
 that simply removed the function call you thought you were measuring.
 
+For instance, let's see this code:
 
-In order to avoid that, it is a good idea to store the result of the function call
+[embedmd]:# (benchmarks_test.go /func div/ /^$/)
+```go
+func div(a, b int) int {
+	return int(float64(a) / float64(b))
+}
+func BenchmarkDiv(b *testing.B) {
+	for i := 0; i < b.N; i++ {
+		div(2, 3)
+	}
+}
+```
+
+If we run the benchmark we will get a surpsiring result:
+
+```bash
+$ go test -bench=Div
+BenchmarkDiv-8   	2000000000	         0.28 ns/op
+PASS
+ok  	github.com/campoy/go-tooling-workshop/3-dynamic-analysis/3-profiling	0.603s
+```
+
+This seems a bit too fast, even for a simple piece of code as the one we're benchmarking.
+We can see the generated code by adding `-gcflags "-S"`:
+
+```bash
+$ go test benchmarks_test.go -bench=Div -gcflags "-S" 2>& out.text
```

Comment:
Can I just clarify, what exactly were you trying to achieve with this redirect? `2>&` would redirect stderr to stdout, but then nothing is happening with stdout. If you intended to direct stdout and stderr to the file out.text you can do this in modern bash with just `&> out.txt` or more traditionally `> out.txt 2>&1`. 

I might be wrong, but running that command gives me:

```
go test benchmarks_test.go -bench=Div -gcflags "-S" 2>& out.text
-bash: out.text: ambiguous redirect
```